daffodil-macro-lib/src/main/scala/org/apache/daffodil/processors/parsers/PointOfUncertaintyMacros.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.processors.parsers
+
+import scala.reflect.macros.blackbox.Context
+
+
+object PointOfUncertaintyMacros {
+
+  /**
+   * For creating a point of uncertainty and ensuring that it is cleaned up
+   * appropriately. This allows the use of standard scala-style syntax to
+   * create and use points of uncertainties, while avoiding allocating
+   * functions. Thus, this allows something like this:
+   *
+   *   pstate.withPointOfUncertainty { pou =>
```

Comment:
So this introduces and binds the new point of uncertainty to this name pou?
Cool. I can see using this technique other places also particularly if it helps avoid allocation, which will make profiling easier.

Reply:
Yep, it took some macro magic to figure out how to get it work, but this technique is definitely easily reusable to avoid allocations.